core/base/projectionFromTable/ProjectionFromTable.h

```diff
@@ -66,22 +59,21 @@ namespace ttk {
       }
 
       // Iterate through each point
-      std::vector<std::vector<double>> coef(
-        noPoints, std::vector<double>(4, 0.0));
+      std::vector<std::array<double, 4>> coef(noPoints);
       for(unsigned int i = 0; i < noPoints; ++i) {
-        std::vector<std::vector<double>> points(4, std::vector<double>(2));
+        std::vector<std::array<double, 2>> points(4);
         for(unsigned int j = 0; j < 4; ++j) {
           points[j][0] = std::get<1>(surfaceValues[quadPoints[i][j]]);
           points[j][1] = std::get<2>(surfaceValues[quadPoints[i][j]]);
         }
 
         // Find barycentric coordinates
-        std::vector<double> tableValues{static_cast<double>(tableXValues[i]),
-                                        static_cast<double>(tableYValues[i])};
-        std::vector<std::vector<double>> trianglePoints(3);
-        std::vector<int> indexes(3);
-        std::vector<double> mid{(points[3][0] - points[1][0]) / 2.0,
-                                (points[0][1] - points[1][1]) / 2.0};
+        std::array<double, 2> tableValues{static_cast<double>(tableXValues[i]),
+                                          static_cast<double>(tableYValues[i])};
+        std::vector<std::array<double, 3>> trianglePoints(3);
```

Comment:
There's something off there... With `std::array` you should not need to resize the outer vector. Maybe you wanted `std::array<std::vector<double>, 3>` instead? Or even `std::array<std::array<double, 3>, 3>`?

Reply:
Indeed, it was needed to use an array of array here, I forgot one. 
Done in the next commit!